### DISCUSSION TITLE
Need context in getStaticPaths for accessing parent slug

### DISCUSSION THREAD

**Discussion by ShivamJoker** [vercel/next.js#20265] [Ideas]
# Feature request
Add parameter of context in `getStaticPaths()` too so that we can access nested path param 
```
─── quiz
    ├── [slug]
    │   └── question
    │       └── [qslug].tsx
    └── [slug].tsx
```

## Is your feature request related to a problem? Please describe.

I want to fetch the question slugs of a particular quiz slug, and for that I will need to know the `slug` in `qslug` page

A clear and concise description of what you want and what your use case is.

I don't know if anyone has occurred this thing but this might be useful in some cases where we need multiple nesting and pre rendering.

## Describe the solution you'd like
I would like to have access to parent slug in the `getStaticPaths`

## Describe alternatives you've considered

Right now I am creating an another API to get all the quiz slugs so that I can fetch it in `getStaticPaths` and map it accordingly.

**Comment by codersgyan** [comment]
I think that will be really helpful.

**Reply by stigkj** [reply]
You can do this now by using the `catch-all-routes` feature, i.e. `/quiz/[...slug].tsx`, ref. https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

Then you can just do the following in `getStaticPaths`:

```
  return {
    paths: [
      { params: { slug: ['a', 'question', 'x'] } },
      { params: { slug: ['b', 'question', 'y'] } },
      { params: { slug: ['c', 'question', 'z'] } },
      ...
    ],
  };
```

As this page would be used for all paths below `/quiz/...`, you would need to return 404 for non-matching URLs, i.e.:

```
  if (<non matching URL>) {
    return {
      notFound: true,
    }
  }
```
